### PR TITLE
Feature/friendships

### PR DESCRIPTION
**Pull request by santiago-rodrig**
# Dear reviewer

This is the implementation of the Comments and Likes as resources in the web app that is intended to be a social network. A summary of the features implemented is next.

- Friendship associations
- Friendship proposals
- Friendship acceptance
- Friendship rejecting
- Friendship canceling
- Personalized feed
- Friends listing
- Friend requests notifications
- Tests for all of the above features

Perhaps I missed something while listing the features that I think were implemented. If there is something wrong, unfinished, or without doing, I'll be waiting for further instructions. To populate the database, use `rails db:seed`, it will take a couple of minutes due to the randomly generated data with Faker and the loops.

**P.D**: to setup PostgreSQL you need to install it, create a user, and set a password. Use this `.env` file template in the app directory. Note that this `.env` file takes into account the `APP_ID` and the `APP_SECRET` of the Facebook app that is needed to be created in order to be able to authenticate users using Facebook.

```shell
POSTGRES_USER='YOUR USER NAME CREATED WITH POSTGRESS'
# by default is the current user name
POSTGRES_PASSWORD='YOUR PASSWORD FOR THE USER CREATED WITH POSTGRES'
POSTGRES_DB='facebook'
POSTGRES_TEST_DB='facebook_test'
APP_ID='THE ID OF YOUR FACEBOOK APP'
APP_SECRET='THE SECRET OF YOUR FACEBOOK APP'
```

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -65,6 +67,73 @@ def comment_post
     redirect_to post_path(@post)
   end
 
+  def friend_requests
```
Comment (mosaaleb) [inline app/controllers/users_controller.rb]: - [ ] **[OPTIONAL]** `users_controller.rb` knows too much and this breaks encapsulation. I would suggest to have different controllers for each functionality. liking, commenting, and friendships method should move to its own controllers.
Please read more here [organizing controllers](http://jeromedalbert.com/how-dhh-organizes-his-rails-controllers/)

File: app/views/users/_user.html.erb [hunk 2]
```diff
@@ -8,6 +8,13 @@
       <dd><%= user.posts.count %></dd>
       <dt>Likes</dt>
       <dd><%= user.total_likes %></dd>
+      <dt>Relation</dt>
+      <dd><mark><%= relation_with(user) %></mark></dd>
     </dl>
+    <% if current_user != user && !current_user.real_friends.include?(user) && !current_user.friends.include?(user) %>
```
Comment (mosaaleb) [inline app/views/users/_user.html.erb]: - [ ] There should be no logic in the views.



Please move this logic into helper method.

File: app/views/users/friend_requests.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<h1 class="mb-3 text-center">Your friend requests</h1>
+
+<div class="row">
+  <% if @requests.any? %>
```
Comment (mosaaleb) [inline app/views/users/friend_requests.html.erb]: - [ ] Same here no logic in views.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (santiago-rodrig) [reply app/controllers/users_controller.rb]: I was thinking the same, is done now.

File: app/views/users/_user.html.erb (hunk 2 above)
Comment (santiago-rodrig) [reply app/views/users/_user.html.erb]: Totally agree.

**Pull request merged**